docs/src/main/asciidoc/mongodb-panache.adoc

```diff
@@ -528,7 +529,41 @@ you need to provide the value by yourself.
 ====
 
 `ObjectId` can be difficult to use if you want to expose its value in your REST service.
-So we created JSON-B and Jackson providers to serialize/deserialize them as a `String` which are are automatically registered if your project depends on one of the RESTEasy with JSON-B or RESTEasy with Jackson extensions.
+So we created JSON-B and Jackson providers to serialize/deserialize them as a `String` which are automatically registered if your project depends on one of the RESTEasy with JSON-B or RESTEasy with Jackson extensions.
+
+== Working with Kotlin Data classes
+
+Kotlin data classes are very practical to defines data carrier classes, so they are a perfect match for an entity class.
+
+But this type of class comes with some limitations: all fields needs to be initialized at construction time or be marked at nullable,
+and the generated constructor need to have as parameters all the fields of the data class.
+
+MongoDB with Panache uses the PojoCodec, this MongoDB codec mandate the presence of an empty (no parameter) constructor.
+
+So, if you want to use a data class for an entity class, you need a way to have Kotlin generates an empty constructor.
+To do so, you need to provide default values for all fields of your classes.
+The following sentence from the Kotlin documentation explain this:
+
+__On the JVM, if the generated class needs to have a parameterless constructor, default values for all properties have to be specified (see Constructors).__
```

Comment:
A link to the doc maybe?

Reply:
Yes, we absolutely should have a link. Also, are you sure this work in native?

Reply:
> are you sure this work in native
The pojo codec works in native, and all classes are register for reflection so I don't see why it will not works in native. I can ask  for tests on the original issue report.

Reply:
The reason I am asking is because I have seen some weird deserilization behavior in #3954 with data classes that only occurs in native mode.

Reply:
The native issue you referes to seems to be more on GraalVM side than on Kotin/Mongodb/Quarkus side ... Maybe someone from GraalVM can have a look at a support on GraalVM for kotlin data classes.

Reply:
Yeah probably. I haven't had time to really dig into it yet or talk to the Kotlin folks first to get their opinion.
Once I do that I'll have a better understanding and then we'll be able to proceed further